sympy/core/numbers.py

```diff
@@ -155,25 +155,30 @@ def igcd(*args):
     5
 
     """
-    a = args[0]
-    for b in args[1:]:
-        try:
-            a = _gcdcache[(a, b)]
-        except KeyError:
-            a, b = as_int(a), as_int(b)
-
-            if a and b:
+    a = abs(as_int(args[0])) if args else 0
+    k = 1
+    if a != 1:
+        while k < len(args):
+            b = args[k]
+            k += 1
+            try:
+                a = _gcdcache[(a, b)]
+            except KeyError:
+                b = as_int(b)
```

Comment:
My thought was this: S(1) in {1:2} is True so the type doesn't matter. The usual use case is to use `igcd` on int/Integers so we can get the answer before doing a function call to `as_int`.